cirq-core/cirq/sim/simulator.py

```diff
@@ -606,14 +606,14 @@ def _base_iterator(
         """
         qubits = ops.QubitOrder.as_qubit_order(qubit_order).order_for(circuit.all_qubits())
         act_on_args = self._create_act_on_args(initial_state, qubits)
-        return self._core_iterator(circuit, act_on_args)
+        return self._core_iterator(circuit, act_on_args, qubits)
 
     @abc.abstractmethod
     def _create_act_on_args(
         self,
         initial_state: Any,
         qubits: Sequence['cirq.Qid'],
-    ) -> TActOnArgs:
+    ) -> Dict[Optional['cirq.Qid'], TActOnArgs]:
```

Comment:
This return type is getting long. I can also imagine some metadata that we might want to pass through or whatever some day. I lean toward making a wrapper class `ActOnArgsContainer` for this.

Reply:
Ordinarily I'd do this in a separate PR, but then it creates backwards compatibility concerns.